test/suite/hook_tiered.py

```diff
@@ -52,35 +60,51 @@
 
 # These are the hook functions that are run when particular APIs are called.
 
+# Add the local storage extension whenever we call wiredtiger_open
+def wiredtiger_open_tiered(ignored_self, args):
+    auth_token = "test_token"
+    bucket = "mybucket"
+    extension_name = "local_store"
+    prefix = "pfx-"
+    # XXX This assumes we are running from WT_TEST/test* in the build directory!
+    extension_library = os.getcwd() + \
+            '/../../ext/storage_sources/local_store/.libs/libwiredtiger_local_store.so'
+
+    if not os.path.exists(bucket):
+        os.mkdir(bucket)
+    tier_string = ',,,tiered_storage=(auth_token=%s,' % auth_token + \
+      'bucket=%s,' % bucket + \
+      'bucket_prefix=%s,' % prefix + \
+      'name=%s),tiered_manager=(wait=0),' % extension_name + \
+      'extensions=[\"%s\"],,,' % extension_library
+
+    args = list(args)         # convert from a readonly tuple to a writeable list
+    args[-1] += tier_string   # Modify the list
+
+    WiredTigerTestCase.verbose(None, 3,
+        '    Calling wiredtiger_open with config = \'{}\''.format(args))
+
+    return args
+
 # Called to replace Session.create
-def session_create_replace(ntiers, orig_session_create, session_self, uri, config):
+def session_create_replace(orig_session_create, session_self, uri, config):
     if config == None:
-        base_config = ""
+        new_config = ""
     else:
-        base_config = config
-
-    # If the test is creating a table (not colstore or lsm), create a tiered table instead,
-    # using arg to determine number of tiers. Otherwise just do the create as normal.
-    #
-    # NOTE: The following code uses the old API for creating tiered tables.  As of WT-7173
-    # this no longer works.  It will be updated and fixed in WT-7440.
-    if (uri.startswith("table:") and "key_format=r" not in base_config and
-      "type=lsm" not in base_config):
-        tier_string = ""
-        for i in range(ntiers):
-            new_uri = uri.replace('table:', 'file:tier' + str(i) + '_')
-            orig_session_create(session_self, new_uri, config)
-            tier_string = tier_string + '"' + new_uri + '", '
-        tier_config = 'type=tiered,tiered=(tiers=(' + tier_string[0:-2] + ')),' + base_config
-        WiredTigerTestCase.verbose(None, 3,
-            'Creating tiered table {} with config = \'{}\''.format(uri, tier_config))
-        ret = orig_session_create(session_self, uri, tier_config)
-    else:
-        ret = orig_session_create(session_self, uri, config)
+        new_config = config
+
+    # If the test isn't creating a table (i.e., it's a colstore or lsm) create it as a
+    # "local only" object.  Otherwise we get tiered storage from the connection defaults.
+    if not uri.startswith("table:") or "key_format=r" in new_config or "type=lsm" in new_config:
+        new_config = new_config + ',tiered_storage=(name=none)'
```

Comment:
It would be nice to just skip tests that we can't run as tiered tables.  There's no point wasting time on them.  `skipTest()` exists for this, but is a method on `WiredTigerTestCase`.  Is there a way to invoke it from within a hook?

Reply:
I agree with the thought but this would be for @ddanderson to answer. I have no idea.